detector.go

```diff
@@ -223,10 +232,11 @@ func (d *Detector) detectGroup(group buildpack.Group, done []buildpack.GroupElem
 
 			if d.PlatformAPI.AtLeast("0.12") {
 				targetMatch := false
-				if isWildcard(d.AnalyzeMD.RunImageTarget()) {
+				if isWildcard(d.AnalyzeMD.RunImageTarget()) || hasWildcard(bpDescriptor.Targets) {
 					targetMatch = true
 				} else {
 					for i := range bpDescriptor.Targets {
+						fmt.Println("Checking for match against descriptor:", bpDescriptor.Targets[i])
```

Comment:
```suggestion
						d.Logger.Debugf("Checking for match against descriptor:", bpDescriptor.Targets[i])
```
(this could be helpful as a debug message)